src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp

```diff
@@ -548,17 +548,39 @@ void QgsCategorizedSymbolRendererWidget::changeSelectedSymbols()
 
 void QgsCategorizedSymbolRendererWidget::changeCategorizedSymbol()
 {
+  QgsPanelWidget *panel = QgsPanelWidget::findParentPanel( this );
   QgsSymbol *newSymbol = mCategorizedSymbol->clone();
-  QgsSymbolSelectorWidget *dlg = new QgsSymbolSelectorWidget( newSymbol, mStyle, mLayer, nullptr );
-  dlg->setContext( mContext );
+  if ( panel && panel->dockMode() )
+  {
+    QgsSymbolSelectorWidget *dlg = new QgsSymbolSelectorWidget( newSymbol, mStyle, mLayer, panel );
```

Comment:
I'd suggest you to use `std::uniqe_ptr<QgsSymbolSelectorWidget>` here and `qgis::make_unique<QgsSymbolSelectorWidget>( newSymbol, mStyle, mLayer, panel );` (and remove the `delete`)

Reply:
Hi @elpaso thanks for the review and sorry for the long silence. 
Sorry also to say that i do understand almost nothing in your message: my c++ knowledge is near zero and this PR is the fruit of some trial and errors (copy-paste) coding. I do not have time nor a build environment to give it another try so i'd really really appreciate any code sample i could just copy paste to comply with the review. Any helper?